tuplex/core/src/ee/local/Sort.cc

```diff
@@ -0,0 +1,605 @@
+// are computed from the row length information since each row should be the same
+// number of bytes.
+static std::vector<PartitionSortType> staticallyComputeOffsets(const std::vector<python::Type>& colTypes, int64_t numRows, int partitionNum = 0) {
+    std::vector<PartitionSortType> offsets;
+    int bytesPerRow = 0;
+    for (const python::Type& colType : colTypes) {
+        bytesPerRow += fixedLengthTypeSerializationSize(colType);
+    }
+    int totalAllocated = 0;
+    for (int i = 0; i < numRows; i++) {
+        auto offset = std::make_pair(totalAllocated, totalAllocated + bytesPerRow);
+        offsets.emplace_back(std::make_pair(std::make_pair(partitionNum, i), offset));
+        totalAllocated += bytesPerRow;
+    }
+    return offsets;
+}
+
+// offsets of the form <offset of beginning of row from start of partition...
+// ..., offset of end of row from start of partition> are computed either
+// statically or dynamically. If the types of each column are a fixed length
+// type, this function will call a helper function to statically generate
+// the offsets. Otherwise, it will can a helper function to dynamically
+// generate the offsets.
+std::vector<PartitionSortType> computeOffsets(const tuplex::Schema& schema, const size_t partitionCapacity, const uint8_t* ptr, int64_t numRows, int partitionNum = 0) {
+    python::Type type = schema.getRowType();
+    std::vector<python::Type> colTypes = type.parameters();
+    auto fixedTypeComparator = [&](const python::Type& type) {
+        return !type.isFixedSizeType();
+    };
+    auto x = std::find_if(colTypes.begin(), colTypes.end(), fixedTypeComparator);
+    if (x != colTypes.end()) {
+        // there are var length types
+        return dynamicallyComputeOffsets(schema, partitionCapacity, ptr, numRows, partitionNum);
+    } else {
+        // no var length types
+        return staticallyComputeOffsets(colTypes, numRows, partitionNum);
+    }
+}
+
+// creates order of columns that a user wants to sort by
+std::vector<size_t> genComparisonOrder(int numColumns, std::vector<size_t> order) {
+    std::vector<size_t> ret;
+    if (order.empty()) {
+        for (int i = 0; i < numColumns; i++) {
+            ret.push_back(i);
+        }
+    } else {
+        return order;
+    }
+    return ret;
+    //    for (int i = 0; i < numColumns; i++) {
+    //        if (std::find(indicesToSort.begin(), indicesToSort.end(), i) == indicesToSort.end()) {
+    //            indicesToSort.push_back(i);
+    //        }
+    //    }
+    //    std::vector<int> ret(indicesToSort.begin(), indicesToSort.end());
+    //    return ret;
+}
+
+std::vector<size_t> genComparisonOrderEnum(int numColumns, std::vector<size_t> orderEnum) {
+    std::vector<size_t> ret;
+    if (orderEnum.empty()) {
+        for (int i = 0; i < numColumns; i++) {
+            ret.push_back(i);
+        }
+    } else {
+        return orderEnum;
+    }
+    return ret;
+    //
+    //    for (int i = 0; i < numColumns; i++) {
```

Comment:
remove commented. Add doxygen comments to all functions. Consider passing vectors via `const vector<...>&`